bool has_func
string func_name
hademo/Args args
